src/server.rs

```diff
@@ -266,6 +267,15 @@ impl ServiceBuilder {
     }
 }
 
+/// Used to indicate the result of the check. If it returns `Abort`,
+/// skip the subsequent checkers and abort the grpc call.
+pub enum CheckResult {
+    Continue,
+    Abort(RpcStatus),
+}
+
+pub(crate) type CheckClosure = Arc<dyn Fn(&RpcContext) -> CheckResult + Send + Sync>;
```

Comment:
Does it have to be `Sync` and `Sync`? How about `Send` and `Clone` just like service object?

Reply:
I found that removing `Send` `Sync` also works. And why need to add `Send` and `Clone`?

Reply:
I'm afraid there are some types not constraint correctly. It will be sent and visit in different threads, so it has to be at least `Send`.